src/relentless/optimize/method.py

```diff
@@ -351,6 +352,7 @@ def optimize(self, objective, design_variables, directory=None):
 
         if directory is not None:
             directory = data.Directory.cast(directory, create=mpi.world.rank_is_root)
+            directory.clear_contents()
```

Comment:
Do we want to have an `overwrite` option here? We could check if `directory` is not empty, then if `overwrite` is True, clear contents; otherwise, raise an error.